cobalt-parser/src/lexer/tokenizer.rs

```diff
@@ -78,13 +74,18 @@ impl<'src> TokenStream<'src> {
         self.0.is_empty()
     }
 }
-
 impl<'src> SourceReader<'src> {
-    pub fn tokenize(&mut self) -> (TokenStream<'src>, Vec<CobaltError<'src>>) {
-        let mut tokens = Vec::new();
-        let mut errors = Vec::new();
-
+    pub fn eat_tokens(
```

Comment:
Could you document what this function returns, and its parameters (particularly `max_idx`)